### DISCUSSION TITLE
BitOps Community Meeting (28 Sep 2022): Survey Results, Twitch Demo ideas, Roadmap

### DISCUSSION THREAD

**Discussion by arm4b** [bitovi/bitops#306] [Announcements]
We invite everyone (developers, adopters, or just curious people) to join the bi-weekly BitOps Community `1 hour` Meeting on Wednesday, `28 Sep 2022, 12:00 PM US East` to discuss the ongoing project developments, updates, and plans, share use cases and stories or ask/answer questions about BitOps tool.

See https://github.com/bitovi/bitops/discussions/288 for more info on how to join.

### Follow-ups from the previous #301 Meeting
- BitOps benefits Brainstorming -> Landing Page
- Parked BitOps Twitter/LN social accounts (for future)
- Update the BitOps StackStorm/HeyEmoji blogs to v2

### BitOps Survey Results Overview & Discussion
Discussion and overview of the Internal (Bitops Engineers) Survey. What we learned, what to improve in BitOps, and how it connects to the Roadmap.

### BitOps Demo on Twitch
Based on @dylan-bitovi idea from the previous meeting https://github.com/bitovi/bitops/discussions/294, we could use Bitovi Twitch and host something like a BitOps demo. Any ideas on what could be a good topic example for a demo, any interested folks hosting it, event format, marketing pre-req, prep work?

### Roadmap https://github.com/bitovi/bitops/discussions/245 & Development Discussions
* Create a Roadmap page (docs)?
* [BitOps Plugin support for private repos #60](https://github.com/bitovi/bitops/issues/60)
* [Parse passwords / credentials out of logging output #208](https://github.com/bitovi/bitops/issues/208)
* [Project Quality](https://github.com/bitovi/bitops/projects/4) as KPI
  * Documentation:
    * #282
    * https://github.com/bitovi/bitops/issues/302
    * https://github.com/bitovi/bitops/issues/303
    * https://github.com/bitovi/bitops/issues/143
    * https://github.com/bitovi/bitops/issues/280
    * Cleanup docs from `TODOs`, remove unimplemented features
  * Testing
    * https://github.com/bitovi/bitops/issues/286
    * https://github.com/bitovi/bitops/issues/271
    * https://github.com/bitovi/bitops/issues/287
  * CI/CD
    * https://github.com/bitovi/bitops/issues/293
    * https://github.com/bitovi/bitops/issues/224
  * Stability/bugfixes
  * Feature consistency
* Ops Repo Composability (needs a dedicated issue w description)
* [Default folder support](https://github.com/bitovi/bitops/issues/177) follow-up

**Comment by mickmcgrath13** [comment]
# in-call notes:

## General questions
- Post notices for the community calls on LinkedIn, too (is happening)
- parked a twitter (bitops_sh)
- bitops.sh

## twitch demo
- options
  - dylan: something simple, tried&true.  an existing blog post
    - terraform, ansible, website create (https://www.bitovi.com/blog/bitops-terraform-ansible)
    - should update for v2
  - phil: maybe the heyemoji one
  - mick: existing TF repo migrate into bitops
- considerations
  - want to minimize IaC (i.e. it's not an IaC tutorial)
    - focus on the power that bitops brings to connect IaC tools
  - maybe a subset of the bitovi devops twitch
  - maybe use an existing blog as a litmus test
- get ourselves on someone else's meetup
  - example: https://twitter.com/luca_cloud
  - Phil's notes: 
    - Leveraging audiences of reputable platforms means that we can tap into a resource that already exists. 
    - This def does not need to be scoped to BitOps as this would be a great way to possibly find new client leads as we prove our DevOps expertise
    - It can also be treated as a handout, specifically with a little "reach out to us" slide


## Survey results discussion 
- continue to focus on "beginners"
- multi-step pipeline flow
  - difficult to debug or rerun if something fails in the middle
  - maybe divide bitops into steps (similar to competitors)
- BitOps is really (maybe too much) focused on CD
  - need a focus on CI
  - could be marketing/docs (ex: PRs that run TF Plan in docs)
- docs enhancements
  - need a getting started guide (look at dagger)
  - look over issues in GitHub
- Terraform (is a bad fit?)
  - plan/apply
  - could be docs enhancements
- env duplication
  - there are already tools for TF (and k8s) that organize envs in a native way
  - terragrunt plugin?
- shared ops repo and blast radius
  - multiple helm deployments
  - mitigation: env-focused pipelines
  - mitigation: only run pipelines for charts that changed
  - mitigation: back-and-forth with "gates"
- too many env variables
  - maybe a docs issue
  - .env file to mitigate the issue of needing to mess with the `deploy.sh` script (could vary based on pipeline runner)
  - maybe also could introduce `bitops.config.yaml` files at the root ops repo level and env level
- MVP and versioning
  - that we're on 2.0.0 can be misleading (maybe versioning is too aggressive
- marketing
  - bitovi org level marketing _could_ be a hinderance to bitops specific marketing
  - we do need more meetups, though
  - just need to find outlets to spread the word here
- project quality and intuitiveness
- missing social proof

## Roadmap
- landing page
- getting started guides & docs
  - example: PRs for TF plan, quick start for ansible, best practices/use cases, blast radius mitigation, helm chart separation
- code quality
- spreading the word

## review landing page notes
didn't talk of this specifically

## Takeaways, next steps
- add the heyemoji slack app youtube video to the blog
  - youtube: https://www.youtube.com/watch?v=TVX_87r-ySA
  - blog: https://www.bitovi.com/blog/heyemoji-slack-app-aws-terraform-ansible
- twitch stream:  get _one_ scheduled for bitops (maybe as a follow up to getting on another person's meetup)
- Heather to start compiling a list of people who does meetups to get onto
  - need a slide deck for BitOps (benefits, anticipated questions, etc)
- Review/organize a roadmap (see above)
